classes/migration/upgrade/v3_4_0/I9813_QuickSubmitSubmissionProgressType.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I9813_QuickSubmitSubmissionProgressType.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I9813_QuickSubmitSubmissionProgressType
+ *
+ * @brief Fix the old submission_progress values inserted by QuickSubmit plugin,
+ *   from an int to a string to match the new step ids
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Exception;
+use Illuminate\Support\Facades\DB;
+use PKP\db\DAORegistry;
+use PKP\install\DowngradeNotSupportedException;
+
+class I9813_QuickSubmitSubmissionProgressType extends \PKP\migration\Migration
+{
+    public function up(): void
+    {
+        $thresholdVersion = '1.0.7.1';
+        /** @var VersionDAO $versionDao */
+        $versionDao = DAORegistry::getDAO('VersionDAO');
+        $installedQuickSubmitPluginVersion = $versionDao->getCurrentVersion('plugins.importexport', 'quickSubmit');
+        if ($installedQuickSubmitPluginVersion->compare($thresholdVersion) === 0) {
```

Comment:
because the script is executed only for OJS versions 3.4.0.0 - 3.4.0.5, it is OK to check for only this one plugin version, right? -- the older versions will not work with 3.4.x...

Reply:
Yes, AFAICS there's just one version of the QuickSubmit for 3.4.